webpack/assets/javascripts/react_app/components/ModelsTable/ModelsTableActions.js

```diff
@@ -0,0 +1,43 @@
+import {
+  TABLE_REQUEST,
+  TABLE_SUCCESS,
+  TABLE_FAILURE,
+  INIT_TABLE,
+} from './ModelsTableConsts';
+import { ajaxRequestAction } from '../../redux/actions/common';
+
+export const updateSearch = (current) => {
+  const init = {
+    page: current.params && current.params.page,
+    perPage: current.params && current.params.per_page,
+    order: current.params && current.params.order,
+    searchQuery: current.params && current.params.search,
+    results: current.table,
+  };
+
+  const payload = Object.keys(init)
+    .filter(k => ![undefined, null].includes(init[k]) && k !== 'order')
```

Comment:
afaik this filtering  shouldn't be in the action, I think it would be better to keep the redux state as is , and then use a selector via `mapsStateToProps` for filtering.

Reply:
SGTM, I am just waiting on [your PR](https://github.com/theforeman/foreman/pull/5778) before doing so.